kernel/poll.c

```diff
@@ -154,7 +154,9 @@ static inline void clear_event_registration(struct k_poll_event *event)
 	if (event->poller == NULL) {
 	        return ;
         }
-
+	
+	event->poller = NULL;
```

Comment:
Why is this in a separate patch?  The state between your two patches is clearly incorrect now, as the field is never set to NULL.